test/gpu/native/atomics.chpl

```diff
@@ -0,0 +1,159 @@
+use GPU, CTypes, GpuDiagnostics, Types, Random;
+
+config const verbose = false;
+config param excludeForRocm;
+
+// CPU versions of associated GPU functions. These aren't actually atomic but
+// are used for test verification:
+proc cpuAtomicAdd (ref x : ?T, val : T) : void { x += val; }
+proc cpuAtomicSub (ref x : ?T, val : T) : void { x -= val; }
+proc cpuAtomicExch(ref x : ?T, val : T) : void { x = val; }
+proc cpuAtomicMin (ref x : ?T, val : T) : void { x = min(x, val); }
+proc cpuAtomicMax (ref x : ?T, val : T) : void { x = max(x, val); }
+proc cpuAtomicInc (ref x : ?T, val : T) : void { x = if x >= val then 0:T else x + 1; }
+proc cpuAtomicDec (ref x : ?T, val : T) : void { x = if x == 0:T || x > val then val else x-1; }
+proc cpuAtomicAnd (ref x : ?T, val : T) : void { x &= val; }
+proc cpuAtomicOr  (ref x : ?T, val : T) : void { x |= val; }
+proc cpuAtomicXor (ref x : ?T, val : T) : void { x ^= val; }
+proc cpuAtomicCAS (ref x : ?T, cmp : T, val : T) : void { x = if x == cmp then val else x; }
+
+proc check(orig, gpuRes, cpuRes, s) {
+  stopGpuDiagnostics();
+  const nLaunch = getGpuDiagnostics()[0].kernel_launch;
+
+  if nLaunch !=1 then
+    writeln(s + " didn't result in correct number of kernel launches. " +
+            nLaunch:string + " launches detected.");
+  else if verbose then
+    writeln(s + " resulted in kernel launch");
+
+  // To get this to work we'll need to resolve https://github.com/chapel-lang/chapel/issues/22114
+  /*var isCorrect = true;
+  for (x,y) in zip(gpuRes,cpuRes) do isCorrect &= isclose(x,y);
+  var printVals = false;
+  if !isCorrect {
+    writeln(s + " computed wrong result. ");
+    printVals = true;
+  } else if verbose {
+    writeln(s + " computed right result. ");
+    printVals = true;
+  }
+  if(printVals) {
+    writeln("input values: ", orig);
+    writeln("gpu result: ", gpuRes);
+    writeln("cpu result: ", cpuRes);
+  }*/
+
+  resetGpuDiagnostics();
+  startGpuDiagnostics();
+}
+
+proc runTest(param op : string, type T) {
+  var origVals : 3*T;
+  var gpuVals  : 3*T;
+  var cpuVals  : 3*T;
+  var rng = createRandomStream(eltType=T, parSafe=false);
+  for i in 0..2 do origVals[i] = rng.getNext();
+  gpuVals = origVals;
+  cpuVals = origVals;
+
+  // It would be better if we could pass 'runTest' the gpu and cpu functions
+  // that we want to test as first class functions instead of having this
+  // repetitive switchyard. Alas, those functions are generic so (as of today)
+  // can not be captured as FCF.
```

Comment:
`check` should check for number of launches. You can drop `assertOnGpu`s to make loops look better if you want.